standards/application/messaging-api.md

```diff
@@ -0,0 +1,944 @@
+
+Messages can be identified by:
+- The `message hash` as described in the [MESSAGE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/14/message.md#deterministic-message-hashing) specification.
+- The `contentTopic` associated with the message, as defined in the [TOPICS](https://github.com/vacp2p/rfc-index/blob/main/waku/informational/23/topics.md#content-topic-format) specification.
+- The `RequestID` described in the `Send API` section.
+- The `SubscriptionID` described in the `Subscribe API` section.
+
+These identifiers SHOULD be passed to the appropriate methods described below to retrieve or manage messages.
+
+Depending on the presence of the `storeConfirmation` or `receiveConfirmation` options in the `Initial configuration`, the `Message Storage API` will trigger either the `Subscribe API` or `Acknowledgements` routine.
+
+Once a message is added to the `Message Storage API`, or if any details about it change,
+the appropriate events from the `message:*` event family (as described in the `Event Source API` section) MUST be invoked after the message is added or updated in the underlying storage.
+
+#### Acknowledgements
+
+Message Acknowledgement mechanism is a background operation that MUST:
+- Perform store‑based reliability checks as specified in [P2P‑RELIABILITY](./p2p‑reliability.md).  
+- Halt when the node’s health status transitions to `unhealthy` (as defined by the `Health API`).  
+- Resume when the node’s health status transitions to either `minimally healthy` or `healthy`.  
+- Set the `stored` or `received` field on `MessageRecord` once the message is acknowledged according to store‑based reliability.
+
+This operation MUST be initiated when any of the following conditions is met:
+- The `storeConfirmation` or `receiveConfirmation` property is `true` and `confirmContentTopics` is provided.  
+- The `Subscribe API` is invoked for one or more content topics.  
+- The `Send API` has successfully sent one or more messages.
+
+For the recurring [STORE](../standards/core/store.md) queries it is RECOMMENDED to prioritize nodes in this order:
+1. `staticStoreNode` specified in the initial configuration.
+2. `preferredServiceNodes`, if provided.
+3. Nodes discovered via network discovery.
+
+#### Programmatic API
+
+```typescript
+type MessageRecord = {
+  sending?: boolean;
+  sent?: boolean;
+  stored: boolean;
+  received: boolean;
+  requestId?: RequestID;
+  message: IWakuMessage;
+  error?: unknown;
+};
+
+interface MessageStorage {
+  getByRequestId(id: RequestID): MessageRecord | undefined;
+  getByHash(hash: string, skip?: number, take?: number): MessageRecord | undefined;
+  getByContentTopic(contentTopic: string, skip?: number, take?: number): MessageRecord[];
+  getBySubscriptionId(id: SubscriptionID, skip?: number, take?: number): MessageRecord[];
+}
+```
+
+#### Properties of `MessageRecord`
+
+##### `sending`
+Defaults to `false`.
```

Comment:
The concept of `default` does not make sense here.
it's either `true` because there's still attempt to send
or `false` because it stops sending due to giving up or being somewhat successful